streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1500,68 +1516,71 @@ private void validateRackAwarenessConfiguration() {
-                                    "consumer", config, eosMessage, clientProvidedProps.get(config), CONSUMER_EOS_OVERRIDES.get(config)));
-                            clientProvidedProps.remove(config);
-                        }
-                    } else if (PRODUCER_EOS_OVERRIDES.containsKey(config)) {
-                        if (!clientProvidedProps.get(config).equals(PRODUCER_EOS_OVERRIDES.get(config))) {
-                            log.warn(String.format(nonConfigurableConfigMessage,
-                                    "producer", config, eosMessage, clientProvidedProps.get(config), PRODUCER_EOS_OVERRIDES.get(config)));
-                            clientProvidedProps.remove(config);
-                        }
-                    } else if (ProducerConfig.TRANSACTIONAL_ID_CONFIG.equals(config)) {
-                        log.warn(String.format(nonConfigurableConfigMessage,
-                            "producer", config, eosMessage, clientProvidedProps.get(config), "<appId>-<generatedSuffix>"));
-                        clientProvidedProps.remove(config);
-                    }
-                }
+    private void validateConsumerPropertyMap(final Map<String, Object> props) {
+        if (eosEnabled) {
+            // Iterate over KS_CONTROLLED_CONSUMER_CONFIGS_EOS_ENABLED and override values if set
+            for (final Map.Entry<String, Object> entry : KS_CONTROLLED_CONSUMER_CONFIGS_EOS_ENABLED.entrySet()) {
+                overwritePropertyMap(props, entry.getKey(), entry.getValue(), "consumer");
+            }
+            verifyMaxInFlightRequestPerConnection(props.get(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION));
+
+        } else {
+            // Iterate over KS_CONTROLLED_CONSUMER_CONFIGS and override values if set
+            for (final Map.Entry<String, Object> entry : KS_CONTROLLED_CONSUMER_CONFIGS.entrySet()) {
+                overwritePropertyMap(props, entry.getKey(), entry.getValue(), "consumer");
             }
         }
+    }
 
+    private void validateProducerPropertyMap(final Map<String, Object> props) {
         if (eosEnabled) {
-            verifyMaxInFlightRequestPerConnection(clientProvidedProps.get(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION));
+            // Iterate over KS_CONTROLLED_CONSUMER_CONFIGS_EOS_ENABLED and override values if set
+            for (final Map.Entry<String, Object> entry : KS_CONTROLLED_PRODUCER_CONFIGS_EOS_ENABLED.entrySet()) {
+                overwritePropertyMap(props, entry.getKey(), entry.getValue(), "producer");
+            }
+            verifyMaxInFlightRequestPerConnection(props.get(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION));
+
+        }
+
+        if (props.containsKey(ProducerConfig.PARTITIONER_CLASS_CONFIG)) {
+            final Class<?> c = (Class<?>) props.get(ProducerConfig.PARTITIONER_CLASS_CONFIG);
+            if (!StreamPartitioner.class.isAssignableFrom(c)) {
```

Comment:
`StreamPartitioner` and the producer `Partitioner` interface are totally independent -- I would never expect a class to implement both interfaces. Not sure what the purpose of this check is?

Reply:
This was an attempt to validate the `PARTITIONER_CLASS_CONFIG`.
But reading the above conversation I believe the decision is to make no change for this config in this PR.

Have removed this change and resolving this comment.